src/pubsub.c

```diff
@@ -475,7 +565,29 @@ int pubsubPublishMessageInternal(robj *channel, robj *message, pubsubtype type)
         return receivers;
     }
 
-    /* Send to clients listening to matching channels */
+    /* Send to clients listening to channels matching prefix patterns */
+    raxDescend d;
+    raxDescendStart(&d,server.pubsub_prefixes,channel->ptr,sdslen(channel->ptr));
+    /* Descend the prefix radix tree to find every level that is a partial match of the channel name. */
+    while (raxDescendNext(&d)) {
+        /* Regenerate the pattern with the star at the end */
+        robj *pattern = patternFromPrefix(d.key, d.key_len);
```

Comment:
We can perhaps store the pattern object along with the list of clients in `server.pubsub_prefixes` to avoid dynamic memory allocations when matching messages against the patterns.

Reply:
That's a good idea.

Reply:
Storing the pattern object in the tree would also allow us to remove the code that reassembles the key from the descent API.

Reply:
But maybe it's nice to still have it given that it's not supposed to dynamically allocate memory in most cases.

Reply:
Implemented in https://github.com/particle-iot/redis/pull/1/commits/ea90e54d025e348cd30c2e8ae19feefbe1582148